go/pkg/libproxy/multiplexed.go

```diff
@@ -249,6 +252,26 @@ func (c *channel) recvClose() {
 	c.c.Broadcast()
 }
 
+// SetReadBuffer sets the size of the operating system's receive buffer associated with the connection.
+// See similar function https://pkg.go.dev/net#IPConn.SetReadBuffer
+func (c *channel) SetReadBuffer(bytes int) error {
```

Comment:
Ha I understand why it was `int` it can be setsockopt, it's bounded and can fail.

SO_RCVBUF
              Sets or gets the maximum socket receive buffer in bytes.
              The kernel doubles this value (to allow space for
              bookkeeping overhead) when it is set using [setsockopt(2)](https://man7.org/linux/man-pages/man2/setsockopt.2.html),
              and this doubled value is returned by [getsockopt(2)](https://man7.org/linux/man-pages/man2/getsockopt.2.html).  The
              default value is set by the
              /proc/sys/net/core/rmem_default file, and the maximum
              allowed value is set by the /proc/sys/net/core/rmem_max
              file.  The minimum (doubled) value for this option is 256.
boucane:~$ cat /proc/sys/net/core/rmem_max
212992